lib/inc/internal/facts/linux/filesystem_resolver.hpp

```diff
@@ -25,6 +26,7 @@ namespace facter { namespace facts { namespace linux {
         void collect_mountpoint_data(data& result);
         void collect_filesystem_data(data& result);
         void collect_partition_data(data& result);
+        void populate_partition_attributes(partition& part, std::string const& device_directory, void* cache, std::map<std::string, std::string> const& mountpoints);
```

Comment:
Yeah, it's really due to device mapper devices; it needs to set the name before populating the attributes.